packages/components/addon/components/hds/reveal/index.hbs

```diff
@@ -0,0 +1,21 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+
+<Hds::DisclosurePrimitive class={{this.classNames}} @isOpen={{@isOpen}} ...attributes>
+  <:toggle as |t|>
+    <Hds::Reveal::Toggle::Button
+      aria-controls={{this.contentId}}
+      @text={{if (and t.isOpen @textWhenOpen) @textWhenOpen @text}}
```

Comment:
I can change it but to me this seems simpler and less convoluted.

@MelSumner @alex-ju do either of you have a preference for how to handle something like this?

Reply:
I'd rather just keep it as is. When I try changing it then I'm having trouble getting the text to change because the tracked state of isOpen doesn't seem to work properly (at least so far in what I tried). Anyway changing it introduces more complexity...

Reply:
OK, I see now why it needs to live in the template, because the `isOpen` in the `Reveal` is technically a `isInitiallyOpen` while the `isOpen` in the `Reveal::Toggle::Button` is actually the toggle state (the fact that both have the same name lead me to this confusion). Leave it as is, then.

Reply:
Do you have suggestions for changing any of the names to help prevent confusion?